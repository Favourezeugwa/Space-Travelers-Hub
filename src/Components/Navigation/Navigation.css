.logo {
    width: 50px;
}

.logo-title {
    margin-left: 10px;
}
