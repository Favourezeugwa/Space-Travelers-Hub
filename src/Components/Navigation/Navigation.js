import Container from 'react-bootstrap/Container';
import Nav from 'react-bootstrap/Nav';
import Navbar from 'react-bootstrap/Navbar';
import './Navigation.css';
// import { LinkContainer } from 'react-router-bootstrap';
import { NavLink } from 'react-router-dom';
import Logo from './planet.png';

function Navigation() {
  return (
    <Navbar bg="light" expand="lg" className="border-bottom py-3">
      <Container>
        <Navbar.Brand href="#home" className="d-flex">
          <img src={Logo} alt="Logo" className="logo" />
          <h2 className="ps-3">Space-Travelers-Hub</h2>
        </Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="ms-auto">
            <Nav.Link href="#rockets">
              <NavLink to="/">Rockets</NavLink>
            </Nav.Link>
            <Nav.Link href="#missions">
              <NavLink to="/mission">Missions</NavLink>
            </Nav.Link>
            <Nav.Link href="#myProfile">
              <NavLink to="/myProfile">My Profile</NavLink>
            </Nav.Link>
          </Nav>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
}

export default Navigation;
