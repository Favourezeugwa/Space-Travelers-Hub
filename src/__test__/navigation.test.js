import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Navigation from '../Components/Navigation/Navigation';

it('Test Navigation renders correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <Navigation />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
