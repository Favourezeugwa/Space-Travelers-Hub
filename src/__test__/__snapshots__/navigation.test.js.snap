// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Navigation renders correctly 1`] = `
<nav
  className="border-bottom py-3 navbar navbar-expand-lg navbar-light bg-light"
>
  <div
    className="container"
  >
    <a
      className="d-flex navbar-brand"
      href="#home"
    >
      <img
        alt="Logo"
        className="logo"
        src="planet.png"
      />
      <h2
        className="ps-3"
      >
        Space-Travelers-Hub
      </h2>
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="ms-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rr-ui-event-key="#rockets"
          disabled={false}
          href="#rockets"
          onClick={[Function]}
        >
          <a
            aria-current="page"
            className="text-primary link active"
            href="/"
            onClick={[Function]}
            style={
              Object {
                "borderBottom": "3px solid #0d6efd",
                "color": "#0d6efd",
                "paddingBottom": "0.25rem",
              }
            }
          >
            Rockets
          </a>
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="#missions"
          disabled={false}
          href="#missions"
          onClick={[Function]}
        >
          <a
            className="text-primary link"
            href="/mission"
            onClick={[Function]}
            style={
              Object {
                "borderBottom": "",
                "color": "",
                "paddingBottom": "",
              }
            }
          >
            Missions
          </a>
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="#myProfile"
          disabled={false}
          href="#myProfile"
          onClick={[Function]}
        >
          <a
            className="text-primary link"
            href="/myProfile"
            onClick={[Function]}
            style={
              Object {
                "borderBottom": "",
                "color": "",
                "paddingBottom": "",
              }
            }
          >
            My Profile
          </a>
        </a>
      </div>
    </div>
  </div>
</nav>
`;
